spyder/plugins/editor/widgets/editorstack/editorstack.py

```diff
@@ -653,7 +657,16 @@ def update_fname_label(self):
         """Update file name label."""
         filename = to_text_string(self.get_current_filename())
         if len(filename) > 100:
-            shorten_filename = u'...' + filename[-100:]
+            metrics = QFontMetrics(
+                self.default_font
+                if self.default_font is not None
+                else self.font()
+            )
+            max_width = 100 * metrics.width('W')
+            elided_filename = metrics.elidedText(
+                filename, Qt.ElideMiddle, max_width
+            )
+            shorten_filename = elided_filename
         else:
             shorten_filename = filename
```

Comment:
```suggestion
            shorten_filename = elided_filename
            self.fname_label.setToolTip(filename)
        else:
            shorten_filename = filename
            self.fname_label.setToolTip("")
```

Show the full filename in the tooltip if it's elided, in case users want to take a look at it.